crates/re_renderer/src/allocator/cpu_write_gpu_read_belt.rs

```diff
@@ -0,0 +1,415 @@
+        //      -> buffers are always owned by the pool and are always Arc.
+        //          This means it not allowed to move the buffer out.
+        //          (We could make them Pin<Arc<>> but this complicates things inside the BufferPool)
+        let write_view = unsafe {
+            std::mem::transmute::<wgpu::BufferViewMut<'_>, wgpu::BufferViewMut<'static>>(write_view)
+        };
+
+        CpuWriteGpuReadBuffer {
+            chunk_buffer: self.buffer.clone(),
+            byte_offset_in_chunk_buffer: start_offset,
+            write_view,
+            unwritten_element_range: 0..num_elements,
+            _type: std::marker::PhantomData,
+        }
+    }
+}
+
+/// Efficiently performs many buffer writes by sharing and reusing temporary buffers.
+///
+/// Internally it uses a ring-buffer of staging buffers that are sub-allocated.
+///
+/// Based on to [`wgpu::util::StagingBelt`](https://github.com/gfx-rs/wgpu/blob/a420e453c3d9c93dfb1a8526bf11c000d895c916/wgpu/src/util/belt.rs)
+/// However, there are some important differences:
+/// * can create buffers without yet knowing the target copy location
+/// * lifetime of returned buffers is independent of the [`CpuWriteGpuReadBelt`] (allows working with several in parallel!)
+/// * use of `re_renderer`'s resource pool
+/// * handles alignment in a defined manner
+///  (see this as of writing open wgpu issue on [Alignment guarantees for mapped buffers](https://github.com/gfx-rs/wgpu/issues/3508))
+pub struct CpuWriteGpuReadBelt {
+    /// Minimum size for new buffers.
+    chunk_size: u64,
+
+    /// Chunks which are CPU write at the moment.
+    active_chunks: Vec<Chunk>,
+
+    /// Chunks which are GPU read at the moment.
+    ///
+    /// I.e. they have scheduled transfers already; they are unmapped and one or more
+    /// command encoder has one or more `copy_buffer_to_buffer` commands with them
+    /// as source.
+    closed_chunks: Vec<Chunk>,
+
+    /// Chunks that are back from the GPU and ready to be mapped for write and put
+    /// into `active_chunks`.
+    free_chunks: Vec<Chunk>,
+
+    /// When closed chunks are mapped again, the map callback sends them here.
+    ///
+    /// Behind a mutex, so that our StagingBelt becomes Sync.
+    /// Note that we shouldn't use SyncSender since this can block the Sender if a buffer is full,
+    /// which means that in a single threaded situation (Web!) we might deadlock.
+    sender: mpsc::Sender<Chunk>,
+
+    /// Free chunks are received here to be put on `self.free_chunks`.
+    receiver: mpsc::Receiver<Chunk>,
+}
+
+impl CpuWriteGpuReadBelt {
+    /// All allocations of this allocator will be aligned to at least this size.
+    ///
+    /// Requiring a minimum alignment means we need to pad less often.
+    /// Also, it has the potential of making memcpy operations faster.
+    ///
+    /// Align to 4xf32. Should be enough for most usecases!
+    pub const MIN_ALIGNMENT: u64 = 16;
```

Comment:
A link to wgpu's alignment sizes table would be nice to help put things into context here.

Reply:
Probably a doc link to `wgpu::MAP_ALIGNMENT` too